compiler/compile/OMRCompilation.cpp

```diff
@@ -1987,6 +2003,17 @@ void OMR::Compilation::switchCodeCache(TR::CodeCache *newCodeCache)
       }
    }
 
+// TODO: Add Enum overrides for this.
+void OMR::Compilation::validateIL()
+   {
+   if (_ilValidator == NULL)
+      self()->failCompilation<TR::CompilationException>("Attempting to validate the IL without the ILValidator being initialized");
+   else
```

Comment:
If this is an unexpected state, prefer `TR_ASSERT_FATAL` over failing a compilation, which could be a silent performance issue.